### PR TITLE
Fix: refresh dropin pm list after applying gift cards

### PR DESCRIPTION
**Pull request by erenbesel**
<fixed>

## Fixed
- Drop-in now updates the payment method list after the shopper applies a gift card, in case the available payment methods change because of the updated payment amount.

</fixed>

### CODE REVIEW

File: Demo/Common/IntegrationExamples/DropIn/DropInAdvancedFlowExample.swift [hunk 1]
```diff
@@ -80,6 +80,7 @@ internal final class DropInAdvancedFlowExample: InitialDataAdvancedFlowProtocol
             } else if let order = response.order,
                       let remainingAmount = order.remainingAmount,
                       remainingAmount.value > 0 {
+                ConfigurationConstants.current.value = remainingAmount.value
```
Comment (goergisn) [inline Demo/Common/IntegrationExamples/DropIn/DropInAdvancedFlowExample.swift]: This will update the values in the UserDefaults right?

Comment (erenbesel) [reply Demo/Common/IntegrationExamples/DropIn/DropInAdvancedFlowExample.swift]: Good catch, I will do this another way.

File: Demo/Configuration.swift [hunk 2]
```diff
@@ -131,7 +131,7 @@ internal struct DemoAppSettings: Codable {
     private static let defaultsKey = "ConfigurationKey"
     
     internal var countryCode: String
-    internal let value: Int
```
Comment (goergisn) [inline Demo/Configuration.swift]: This can be reverted right?

**Review by descorp — APPROVED** [review]

File: Demo/Configuration.swift (hunk 2 above)
Comment (erenbesel) [reply Demo/Configuration.swift]: Yes! done

**Review by goergisn — APPROVED** [review]

**Pull request merged**